### PR TITLE
New launchd key for associating a bundle identifier with Login Items

### PR DESCRIPTION
**Pull request by chilcote**
## Description
Example of what it will look like in a plist

```
<key>AssociatedBundleIdentifiers</key>
<array>
    <string>bundle.id.of.app.signed.by.same.team</string>
</array>
```

Apple docs: https://developer.apple.com/documentation/servicemanagement/updating_helper_executables_from_earlier_versions_of_macos

## Related Issue
13287

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by johnmccrae** [comment]
@chilcote - Joseph, thanks for this. Waiting on the docs team. Would you mind adding a DCO to this too? Ian is supposed to be back today. I'll see what we can get done and get this merged before I ship the full Chef-18 release

**Review by jaymzh — DISMISSED** [review]
Sending back to author - need DCO (sign-off) before we can merge.

**Comment by chilcote** [comment]
did I do that right?

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**
